upf/solvers/grounder.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2021 AIPlan4EU project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+from typing import Dict, List,Tuple
+
+import upf.environment
+import upf.solvers as solvers
+import upf.transformers
+from upf.model import FNode, Problem, ProblemKind, Action
+
+
+
+class Grounder(solvers.solver.Solver):
+    """Performs grounding."""
+    def __init__(self, **options):
+        pass
+
+    def ground(self, problem: 'upf.model.Problem') -> Tuple[Problem, Dict[Action, Tuple[Action, List[FNode]]]]:
```

Comment:
We can try to change the interface in this way:
```suggestion
    def ground(self, problem: 'upf.model.Problem') -> Tuple[Problem, Callable[[Plan], Plan]]:
```